ui/src/app/pages/login/sign-in.tsx

```diff
@@ -403,23 +414,33 @@ export class SignInImpl extends React.Component<SignInProps, SignInState> {
     }
   }
 
-  private onSubmit = async () => {
+  // This method is being used if user is international
+  private updateProfileAndCreateAccount = (profile: Profile) => {
+    this.setState({ profile: profile });
+    // setState is flaky, sometimes it is unable to set profile by the time submit happens
+    // hence to be sure lets just pass profile to submit method
+    this.onSubmit(profile);
+  };
+  private onSubmit = async (profileArg = null) => {
     const { enableCaptcha } = serverConfigStore.get().config;
     this.setState({
       loading: true,
     });
     this.props.showSpinner();
 
+    const profileToSubmit =
+      !!profileArg && profileArg.address ? profileArg : this.state.profile;
```

Comment:
Instead, `profileArg` should be non-nullable and pass it every use.

Reply:
This line, which is pretty complex, can be removed if profileArg is always passed.